tensorboard/backend/event_processing/sqlite_writer.py

```diff
@@ -0,0 +1,425 @@
+      run_id INTEGER NOT NULL,
+      inserted_time REAL,
+      started_time REAL,
+      finished_time REAL,
+      run_name TEXT
+    )
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS RunIdIndex
+    ON Runs (run_id)
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS RunNameIndex
+    ON Runs (experiment_id, run_name)
+    WHERE run_name IS NOT NULL
+    """,
+    """
+    CREATE TABLE IF NOT EXISTS Experiments (
+      rowid INTEGER PRIMARY KEY,
+      user_id INTEGER,
+      experiment_id INTEGER NOT NULL,
+      inserted_time REAL,
+      started_time REAL,
+      is_watching INTEGER,
+      experiment_name TEXT
+    )
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS ExperimentIdIndex
+    ON Experiments (experiment_id)
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS ExperimentNameIndex
+    ON Experiments (user_id, experiment_name)
+    WHERE experiment_name IS NOT NULL
+    """,
+    """
+    CREATE TABLE IF NOT EXISTS Users (
+      rowid INTEGER PRIMARY KEY,
+      user_id INTEGER NOT NULL,
+      inserted_time REAL,
+      user_name TEXT,
+      email TEXT
+    )
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS UserIdIndex
+    ON Users (user_id)
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS UserNameIndex
+    ON Users (user_name)
+    WHERE user_name IS NOT NULL
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS UserEmailIndex
+    ON Users (email)
+    WHERE email IS NOT NULL
+    """,
+    """
+    CREATE TABLE IF NOT EXISTS Graphs (
+      rowid INTEGER PRIMARY KEY,
+      run_id INTEGER,
+      graph_id INTEGER NOT NULL,
+      inserted_time REAL,
+      graph_def BLOB
+    )
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS GraphIdIndex
+    ON Graphs (graph_id)
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS GraphRunIndex
+    ON Graphs (run_id)
+    WHERE run_id IS NOT NULL
+    """,
+    """
+    CREATE TABLE IF NOT EXISTS Nodes (
+      rowid INTEGER PRIMARY KEY,
+      graph_id INTEGER NOT NULL,
+      node_id INTEGER NOT NULL,
+      node_name TEXT,
+      op TEXT,
+      device TEXT,
+      node_def BLOB
+    )
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS NodeIdIndex
+    ON Nodes (graph_id, node_id)
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS NodeNameIndex
+    ON Nodes (graph_id, node_name)
+    WHERE node_name IS NOT NULL
+    """,
+    """
+    CREATE TABLE IF NOT EXISTS NodeInputs (
+      rowid INTEGER PRIMARY KEY,
+      graph_id INTEGER NOT NULL,
+      node_id INTEGER NOT NULL,
+      idx INTEGER NOT NULL,
+      input_node_id INTEGER NOT NULL,
+      input_node_idx INTEGER,
+      is_control INTEGER
+    )
+    """,
+    """
+    CREATE UNIQUE INDEX IF NOT EXISTS NodeInputsIndex
+    ON NodeInputs (graph_id, node_id, idx)
+    """,
+]
+
+
+# Defines application ID as hexspeak for "TBOARD[0]", with an expansion digit.
+# This differs from 0xFEEDABEE used in schema.h, which unfortunately exceeds
```

Comment:
What is the impact of this difference? 
```
[...] can be set by the PRAGMA application_id command in order to identify the database as belonging to or associated with a particular application
```

Hmm is the check enforced at the application level or is it done by sqlite?

Reply:
There's no impact/enforcement at all, it's essentially an arbitrary int of metadata that the application can use if it wants to be able to identify a SQLite file as one that it generated.  We don't consume it anywhere today.  It just seemed like good practice to set one, and I noticed the one chosen previously resolves to 0 which makes it useless, so I picked a different one.